File: problems/11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
const maxArea = (height) => {
    let result = 0;
    let left = 0;
    let right = height.length - 1;
    while (left < right) {
        const aria = (right - left) * Math.min(height[left], height[right]);
        result = Math.max(result, aria)
        height[left] > height[right] ? right-- : left++;
    }
    return result;
};

File: problems/11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        result = 0
        while l < r:
            aria = min(height[l], height[r]) * (r - l)
            result = max(result, aria)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return result
